From bdeafb7bc4857e80dbca5192a751eedcf7b69abd Mon Sep 17 00:00:00 2001
From: Luka Perkov <luka@openwrt.org>
Date: Mon, 27 Oct 2014 21:49:46 +0100
Subject: [PATCH] utils: remove unnecessary check of mystat.st_dev

The check is not needed and it breaks lxc-destroy when container is installed
on top of overlayfs. More information why this is a problem on overlayfs can be
found here:

https://kernel.googlesource.com/pub/scm/linux/kernel/git/mszeredi/vfs/+/overlayfs.current/Documentation/filesystems/overlayfs.txt

Signed-off-by: Luka Perkov <luka@openwrt.org>
---
 src/lxc/utils.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- a/src/lxc/utils.c
+++ b/src/lxc/utils.c
@@ -47,7 +47,7 @@
 
 lxc_log_define(lxc_utils, lxc);
 
-static int _recursive_rmdir_onedev(char *dirname, dev_t pdev,
+static int _recursive_rmdir_onedev(char *dirname,
 				   const char *exclude, int level)
 {
 	struct dirent dirent, *direntp;
@@ -108,10 +108,8 @@ static int _recursive_rmdir_onedev(char
 			failed=1;
 			continue;
 		}
-		if (mystat.st_dev != pdev)
-			continue;
 		if (S_ISDIR(mystat.st_mode)) {
-			if (_recursive_rmdir_onedev(pathname, pdev, exclude, level+1) < 0)
+			if (_recursive_rmdir_onedev(pathname, exclude, level+1) < 0)
 				failed=1;
 		} else {
 			if (unlink(pathname) < 0) {
@@ -147,7 +145,7 @@ extern int lxc_rmdir_onedev(char *path,
 		return -1;
 	}
 
-	return _recursive_rmdir_onedev(path, mystat.st_dev, exclude, 0);
+	return _recursive_rmdir_onedev(path, exclude, 0);
 }
 
 static int mount_fs(const char *source, const char *target, const char *type)
